lms/djangoapps/courseware/tests/test_entrance_exam.py

```diff
@@ -124,13 +130,55 @@ def setUp(self):
             user,
             self.entrance_exam
         )
-        self.entrance_exam.is_entrance_exam = True
-        self.entrance_exam.in_entrance_exam = True
         self.course.entrance_exam_enabled = True
         self.course.entrance_exam_minimum_score_pct = 0.50
         self.course.entrance_exam_id = unicode(self.entrance_exam.scope_ids.usage_id)
         modulestore().update_item(self.course, user.id)  # pylint: disable=no-member
 
+        self.client.login(username=self.request.user.username, password="test")
+        CourseEnrollment.enroll(self.request.user, self.course.id)
+
+    @mock.patch('xmodule.x_module.XModuleMixin.has_dynamic_children', mock.Mock(return_value='True'))
+    def test_view_redirect_if_entrance_exam_required(self):
+        """
+        Unit Test: if entrance exam is required. Should return a redirect.
+        """
+        url = reverse('courseware', kwargs={'course_id': unicode(self.course.id)})
+        expected_url = reverse('courseware_section',
+                               kwargs={
+                                   'course_id': unicode(self.course.id),
+                                   'chapter': self.entrance_exam.location.name,
+                                   'section': self.exam_1.location.name
+                               })
+        resp = self.client.get(url)
+        self.assertRedirects(resp, expected_url, status_code=302, target_status_code=200)
+
+    @patch.dict('django.conf.settings.FEATURES', {'ENTRANCE_EXAMS': False})
+    def test_entrance_exam_content_absence(self):
+        """
+        Unit Test: If entrance exam is not enabled then its content e.g. problems should not be loaded.
+        """
+        url = reverse('courseware', kwargs={'course_id': unicode(self.course.id)})
+        resp = self.client.get(url)
+        self.assertNotIn('Exam Problem - Problem 1', resp.content)
+        self.assertNotIn('Exam Problem - Problem 2', resp.content)
+
+    def test_entrance_exam_requirement_message(self):
+        """
+        Unit Test: entrance exam requirement message should be present in response
+        """
+        url = reverse(
+            'courseware_section',
+            kwargs={
+                'course_id': unicode(self.course.id),
+                'chapter': self.entrance_exam.location.name,
+                'section': self.exam_1.location.name
+            }
+        )
+        resp = self.client.get(url)
+        self.assertEqual(resp.status_code, 200)
+        self.assertIn('In order to access the rest of the course, you need to attain', resp.content)
+
```

Comment:
@marcotuts I want to conform one thing here. In this PR i am hiding the message after successfully achieving the exam milestone or completion the exam. 
do you require to show the text after exam completion instead of hiding the text ? please conform.